pcdsdevices/tests/test_signal.py

```diff
@@ -60,6 +60,34 @@ def test_avg_signal():
     assert cb.called
 
 
+def test_avg_signal_with_duration():
+    logger.debug("test_avg_signal_with_duration")
+    sig = Signal(name='raw')
+    avg = AvgSignal(sig, 2, name='avg', duration=2)
```

Comment:
It took me squinting at this and re-reading the code to realize that the duration only does anything when calling `.trigger()`.  (this could be obvious to most people, but I'm glad there were tests to scrutinize.

Reply:
I think maybe this isn't so obvious in general. If you ponder for a bit you're realize that the class is continually accumulating values, so a time duration must only apply in some specific circumstances. I tried to document it appropriately.